aws/resource_aws_lambda_function.go

```diff
@@ -47,22 +48,57 @@ func resourceAwsLambdaFunction() *schema.Resource {
 			"filename": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"s3_bucket", "s3_key", "s3_object_version"},
+				ConflictsWith: []string{"s3_bucket", "s3_key", "s3_object_version", "image_uri"},
```

Comment:
Maybe a good chance to substitute with `ExactlyOneOf: []string{"filename", "s3_bucket", "image_uri"}` this will save the validation check post plan time below.

Reply:
Good idea. We should probably also set `RequiredWith: []string{"s3_key"}` on `s3_bucket`